fgpyo/sam/__init__.py

```diff
@@ -590,6 +587,70 @@ def set_pair_info(r1: AlignedSegment, r2: AlignedSegment, proper_pair: bool = Tr
     r2.template_length = -insert_size
 
 
+@attr.s(auto_attribs=True, frozen=True)
+class ReadEditInfo:
+    """Counts various stats about how a read compares to a reference sequence."""
+
+    mismatches: int
+    insertions: int
+    inserted_bases: int
+    deletions: int
+    deleted_bases: int
+    nm: int
+
+
+def calc_edit_info(
+    r: AlignedSegment, reference_sequence: str, reference_offset: Optional[int] = None
+) -> ReadEditInfo:
```

Comment:
I'd argue for using longer names for parameters and try to avoid abbreviations for function names, where possible.  I think the function name here is less important, but I think `r` needs to be changed at least.


```suggestion
def build_edit_info(
    recrod: AlignedSegment, reference_sequence: str, reference_offset: Optional[int] = None
) -> ReadEditInfo:
```